datalad/distribution/install.py

```diff
@@ -310,90 +162,144 @@ def __call__(
+            annex_opts=annex_opts,
+            reckless=reckless,
+            jobs=jobs,
+        )
+
+        # switch into scenario without --source:
+        if source is None:
+            # we need to collect URLs and paths
+            to_install = []
+            to_get = []
+            for urlpath in path:
+                ri = RI(urlpath)
+                (to_get if isinstance(ri, PathRI) else to_install).append(urlpath)
+
+            installed_items = []
+            common_kwargs['dataset'] = dataset
+
+            # first install, and then get
+            for s in to_install:
+                lgr.debug("Install passes into install source=%s", s)
+                result = Install.__call__(
+                                source=s,
                                 description=description,
-                                recursive=recursive,
-                                recursion_limit=recursion_limit,
-                                save=save,
                                 if_dirty=if_dirty,
-                                git_opts=git_opts,
+                                save=save,
                                 git_clone_opts=git_clone_opts,
-                                annex_opts=annex_opts,
-                                annex_init_opts=annex_init_opts
+                                annex_init_opts=annex_init_opts,
+                                **common_kwargs
                             )
-
-                            installed_items += assure_list(result)
-                        except Exception:
-                            # Note: We don't exactly know what was skipped but
-                            # the `path` requested to be installed, since it will be
-                            # resolved only within the recursive call of install.
-                            lgr.info("Installation of {0} skipped.".format(p))
-
-                    if len(installed_items) == 1:
-                        return installed_items[0]
+                installed_items += assure_list(result)
+
+            if to_get:
+                lgr.debug("Install passes into get %d items", len(to_get))
+                # all commented out hint on inability to pass those options
+                # into underlying install-related calls.
+                # Also need to pass from get:
+                #  annex_get_opts
+                content_by_ds = Get.__call__(
+                    to_get,
+                    # description=description,
+                    # if_dirty=if_dirty,
+                    # save=save,
+                    # git_clone_opts=git_clone_opts,
+                    # annex_init_opts=annex_init_opts
+                    _return_datasets=True,
+                    **common_kwargs
+                )
+
+                # compose content_by_ds into result
+                for dspath in content_by_ds:
+                    ds_ = Dataset(dspath)
+                    if ds_.is_installed():
```

Comment:
Consider it a "defensive (or paranoid) programming" approach. This was done to reduce number of assumptions on what `get` could possibly return.    But I didn't feel too strongly about it -- thus just a warning not an assert ;)  If situation never happens -- no warning would never be spit out.  But if happens -- there would be a warning.